File: Problem Sets/pset6/credit/credit.py
# Get card number
card = input("Number: ")
l = len(card)
start = card[0:2]
card = int(card)
multi = 0
non = 0

# Check whether card is of correct length
if l not in [13, 15, 16]:
    print("INVALID")

while card > 0:

    # Add to non multi
    non += card % 10
    card //= 10

    # Add to multi
    product = (card % 10) * 2
    while product > 0:
        multi += product % 10
        product //= 10

    card //= 10

if (multi + non) % 10 == 0:
    if start in ["34", "37"]:
        print("AMEX")
    elif start in ["51", "52", "53", "54", "55"]:
        print("MASTERCARD")
    elif start[0] == "4":
        print("VISA")
    else:
        print("INVALID")
else:
    print("INVALID")

File: Problem Sets/pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Declaring variables
    long card_num = get_long("Credit card number: ");
    long start_num = card_num;
    int first = 0;
    int multi = 0, not_multi = 0;
    int product;


    // Applying the Luh's Algorithm
    while (card_num > 0)
    {
        // Adding every other number that we do not multiply
        not_multi += card_num % 10;
        card_num /= 10;

        if (card_num < 100 && card_num > 10)
        {
            first += card_num;
        }

        // Multiplying every other number by two and adding digits of calculated products
        product = (card_num % 10) * 2;
        while (product > 0)
        {
            multi += product % 10;
            product /= 10;
        }
        card_num /= 10;

        if (card_num < 100 && card_num > 10)
        {
            first += card_num;
        }
    }

    // Checking whether the card is valid and if it is what company's card it is
    if ((multi + not_multi) % 10 == 0 && start_num > 1000000000000)
    {
        if (first == 34 || first == 37)
        {
            printf("AMEX\n");
        }

        else if (first == 51 || first == 52 || first == 53 || first == 54 || first == 55)
        {
            printf("MASTERCARD\n");
        }

        else if (first >= 40 && first < 50)
        {
            printf("VISA\n");
        }

        // If no requirements are met, return invalid
        else
        {
            printf("INVALID\n");
        }
    }

    // If no requirements are met, return invalid
    else
    {
        printf("INVALID\n");
    }

}